### ISSUE TITLE
Hide "Custom fields" information

### ISSUE THREAD

**Issue by perry-mitchell** [buttercup/buttercup-desktop#580]
Currently we separate custom fields:

![image](https://user-images.githubusercontent.com/3869469/37592145-836bdfd2-2b76-11e8-92de-77adca11dee9.png)

But as we're moving to #540 displaying entry types (Credit card, bank details, SSH key, Note etc.), it'd be good to go to a more generic layout perhaps.

**Comment by perry-mitchell** [comment]
What do you think @sallar? This has been brought up earlier, and now thinking about it, I'm definitely starting to like it as it makes moving to the entry type facades a bit easier..

**Comment by sallar** [comment]
@perry-mitchell Yes I agree. So you mean just showing everything below each other or having different separators?

**Comment by darsain** [comment]
Just show everything below each other. These separators have no semantic value to the user, only to developers. I don't need to know how these fields are implemented under the hood. I just want to see the fields for my item, which are all important, so any separator would be arbitrary and make no sense :)

**Comment by perry-mitchell** [comment]
I would think that showing it altogether, personally, would make it a bit tidier. This is more from the perspective that when we have custom field layouts supported via facades, it’ll be necessary to group them all anyway.

Right now some fields like username and password cannot be removed so it makes sense that they’re at the top and custom fields below. When we use facades the built in facade properties won’t be removable so they’ll also probably be at the top..

**Comment by darsain** [comment]
@perry-mitchell that sound horrible. What you've just described is so inflexible it'll make me probably stay with my current password manager.

Here is where I'm coming from:

I want to use buttercup to also store license keys, GPG keys, and similar stuff. For this I need only **title**, *maybe* **username**, and **license**/**key** field (that supports new-line characters, lack of which is why I'm currently still not using buttercup). It's already bad that I can't remove name & password fields which will be useless and irrelevant with this kind of item, but if you'll make even more fields mandatory, well that's just polluting the UI and nobody's OCD would like that.

How about a compromise: you display all your built in property fields when editing, but hide all empty fields when just viewing the item?

And regarding the dividers again, lets take a license key for Sublime Text as an example. It needs **title**, **username**, and **key** fields. If I understand the proposed architecture correctly, the **key** field would be custom, so there would be a divider between **username** and **key**, which makes no sense to me :( In other words, dividers impose arbitrary semantic divisions that won't fit all types of items people might want to enter into their vault.

And if you won't allow removing, or alternatively won't hide empty/unused stuff, there would also be tons of empty fields between the **username** and **key**, which is completely pointless.

**Comment by perry-mitchell** [comment]
> that sound horrible.

I disagree - and please, tone it down a little. We're not trying to make an uber strict interface, but we need to be mindful of the current architecture and our plans on what we expect users to want out of an interface.

> I want to use buttercup to also store license keys, GPG keys, and similar stuff. For this I need only title, maybe username, and license/key field

We plan to have facades around all of these. Having mandatory fields that cannot be **removed** (of course they can be empty) helps keep expected data mapped to their respective property (eg. A credit card may or may not have a CCV number, but the field should remain so that it's a visual entry).

> It's already bad that I can't remove name & password fields which will be useless and irrelevant with this kind of item

Which is relevant to what I mentioned:

> Right now some fields like username and password cannot be removed

Which may make for a confusing idea, but it's been explained previously. Right now they're mandatory, but when we implement facades they'll be present only in items that require them. The facades are designed to make the interface more flexible, not less.

> take a license key for Sublime Text as an example. It needs title, username, and key fields

>  so there would be a divider between username and key

That's what this task would fix, as a first point of call. The facades to make for more types of entries will come with #540.

**Comment by perry-mitchell** [comment]
To better communicate what this will change:

 * Remove the divider between **mandatory** and **custom** fields

And what #540 will change:

 * Hide Username/Password required fields, along with some custom fields, under a facade
 * Entries of certain types will only show the fields relevant to them
   * If a user creates a key entry, it may be just a title and the key data field by default, for example
 * Custom fields can be added later by the user
